packages/google_mobile_maps/lib/platform_overlay.dart

```diff
@@ -0,0 +1,212 @@
+  /// [deactivateOverlay].
+  void activateOverlay() {
+    _activationCount += 1;
+    if (_activationCount == 1) {
+      SchedulerBinding.instance.addPostFrameCallback((_) {
+        if (_disposed) {
+          return;
+        }
+        final RenderObject object = _context?.findRenderObject();
+        Offset offset;
+        if (object is RenderBox) {
+          offset = object.localToGlobal(Offset.zero) * window.devicePixelRatio;
+        } else {
+          offset = Offset.zero;
+        }
+        showOverlay(offset);
+      });
+    }
+  }
+
+  /// Prevent activating the overlay until a matching call to [activateOverlay].
+  void deactivateOverlay() {
+    _activationCount -= 1;
+    if (_activationCount == 0) {
+      hideOverlay();
+    }
+  }
+
+  @override
+  void didPush(Route<dynamic> route, Route<dynamic> previousRoute) {
+    _currentRoute = route;
+    _previousRoute = previousRoute;
+    if (previousRoute != null && identical(previousRoute, _routeWithOverlay)) {
+      deactivateOverlay();
+      _deactivatedByPush = true;
+    }
+  }
+
+  @override
+  void didPop(Route<dynamic> route, Route<dynamic> previousRoute) {
+    _currentRoute = previousRoute;
+    _previousRoute = route;
+    if (identical(route, _routeWithOverlay)) {
+      deactivateOverlay();
+    } else if (identical(previousRoute, _routeWithOverlay) &&
+        _deactivatedByPush) {
+      _activateOverlayAfterPopAnimations(route, previousRoute);
+    }
+  }
+
+  void _activateOverlayAfterPopAnimations(
+      Route<dynamic> route, Route<dynamic> previousRoute) {
+    if (route is ModalRoute && previousRoute is ModalRoute) {
+      _doOnceAfter(route.animation, () {
+        _doOnceAfter(previousRoute.secondaryAnimation, () {
+          activateOverlay();
+          _deactivatedByPush = false;
+        });
+      });
+    } else if (route is ModalRoute) {
+      _doOnceAfter(route.animation, () {
+        activateOverlay();
+        _deactivatedByPush = false;
+      });
+    } else if (previousRoute is ModalRoute) {
+      _doOnceAfter(previousRoute.secondaryAnimation, () {
+        activateOverlay();
+        _deactivatedByPush = false;
+      });
+    }
+  }
+
+  void _activateOverlayAfterPushAnimations(
+      Route<dynamic> route, Route<dynamic> previousRoute) {
+    if (route is ModalRoute && previousRoute is ModalRoute) {
+      _doOnceAfter(route.animation, () {
+        _doOnceAfter(previousRoute.secondaryAnimation, () {
+          activateOverlay();
+        });
+      });
+    } else if (route is ModalRoute) {
+      _doOnceAfter(route.animation, () {
+        activateOverlay();
+      });
+    } else if (previousRoute is ModalRoute) {
+      _doOnceAfter(previousRoute.secondaryAnimation, () {
+        activateOverlay();
+      });
+    }
+  }
+
+  void _doOnceAfter(Animation<dynamic> animation, void onDone()) {
+    void listener() {
+      if (animation.status != AnimationStatus.forward &&
```

Comment:
This is a bit hard to read.
Would it not be better to check positively_
```
if (animation.status == AnimationStatus.completed ||
  animation.status == AnimationStatus.dismissed) {...}
```

Reply:
Done.